### PR TITLE
Temp fix to treat CURRENT_DATE as expression for column editor

### LINKED ISSUE #9096: Table Editor doesn't allow default values like `CURRENT_DATE`
# Bug report

## Describe the bug

The Table Editor in Studio doesn't allow default values like `CURRENT_DATE`. However, it works as expected when done via SQL (both creating and altering a table).

## To Reproduce

1. Create a new table:

```sql
CREATE TABLE testdate (
    id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
    date DATE NOT NULL
);
```

2. Go to Table Editor and add default value `CURRENT_DATE` to column `date`.

This results in the notification: Failed to update column "date": failed to update pg.columns with the given ID: invalid input syntax for type date: "CURRENT_DATE"

## Expected behavior

When using SQL directly, you can set the default value and it works as expected:

```sql
ALTER TABLE testdate ALTER COLUMN date SET DEFAULT CURRENT_DATE;
```

I expect that the Table Editor supports values like `CURRENT_DATE`. (I have not tested `CURRENT_TIME` and similar, but expect the same behavior)

### PR DESCRIPTION
**Pull request by joshenlim**
Resolves https://github.com/supabase/supabase/issues/9096

### CODE REVIEW

**Comment by saltcod** [comment]
Looks good!  Should we add a note here? 

"Can either be a literal or an expression (e.g uuid_generate_v4(), NOW(), CURRENT_DATE)"

CleanShot 2022-09-30 at 11 26 50@2x

**Review by saltcod — APPROVED** [review]

**Comment by joshenlim** [comment]
> Looks good! Should we add a note here? 
> "Can either be a literal or an expression (e.g uuid_generate_v4(), NOW(), CURRENT_DATE)"

Hmm think let's leave this out for now - imo makes the side panel a little too wordy. The descriptions here should be concise and any further examples/explanations should have an ext link to some docs. i've half a mind to remove "uuid_generate_v4()" too tbh, but haven't found any good official docs link to show what SQL expressions are yet
